site/docs/develop/java.md

```diff
@@ -17,42 +17,202 @@ Maven. The coordinates are:
+### Fetching Contents
+
+Fetches the contents for a single `ContentsKey`
+```java
+ContentsKey key = ContentsKey.of("table.name.space", "name");
+Map<ContentsKey, Contents> map = api.getContents().key(key).refName("dev").get();
+```
+
+Fetches the contents for multiple `ContentsKey`s
+```java
+List<ContentsKey> keys =
+  Arrays.asList(
+  ContentsKey.of("table.name.space", "name1"),
+  ContentsKey.of("table.name.space", "name2"),
+  ContentsKey.of("table.name.space", "name3"));
+Map<ContentsKey, Contents> allContents = api.getContents().keys(keys).refName("dev").get();
+```
 
-### New Backing Store
 
-Nessie comes pre-packaged with three storage engines: JGit, In-Memory and DynamoDB. 
-For production deployments, DynamoDB is the recommended storage engine. The Nessie 
-team is always evaluating new potential backing stores. If you're looking to implement 
-something new, shout out to us via one of our [communication channels](index.md) to 
-discuss what you're thinking so we can figure out the best way to move forward. 
+### Fetching the Commit Log
+
+Fetches the commit log for the `dev` reference
+```java
+LogResponse log = api.getCommitLog().refName("dev").get();
+```
+
+### Fetching Entries
+
+Fetches the entries for the `dev` reference
+```java
+EntriesResponse entries = api.getEntries().refName("dev").get();
+```
+
+### Merging
+
+This merges `fromBranch` into the given `intoBranch`
+```java
+api.mergeRefIntoBranch()
+  .branchName("intoBranch")
+  .hash(intoBranchHash)
+  .fromRefName("fromBranch")
+  .fromHash(fromHash)
+  .merge();
+```
+
+### Transplanting
+
+Transplant/cherry-pick a bunch of commits from `main` into the `dev` branch
+```java
+Branch dev = ...
+api.transplantCommitsIntoBranch()
+    .branchName(dev.getName())
+    .hash(dev.getHash())
+    .fromRefName("main")
+    .hashesToTransplant(Collections.singletonList(api.getReference().refName("main").get().getHash()))
+    .transplant()
+```
+
+
+## Authentication
+
+Nessie has multiple `NessieAuthenticationProvider` implementations that allow different client authentication mechanisms as can be seen below.
+The documentation for how to configure Nessie server authentication can be found [here](../try/authentication.md).
+
+The `BasicAuthenticationProvider` allows connecting to a Nessie server that has `BASIC` authentication enabled.
+```java
+NessieApiV1 api =
+  HttpClientBuilder.builder()
+  .withUri(URI.create("http://localhost:19121/api/v1"))
+  .withAuthentication(BasicAuthenticationProvider.create("my_username", "very_secret"))
+  .build(NessieApiVersion.V_1, NessieApiV1.class);
+```
+
+The `BearerAuthenticationProvider` allows connecting to a Nessie server that has `BEARER` authentication enabled.
+```java
+NessieApiV1 api =
+  HttpClientBuilder.builder()
+  .withUri(URI.create("http://localhost:19121/api/v1"))
+  .withAuthentication(BearerAuthenticationProvider.create("bearerToken"))
```

Comment:
Just realized that we probably need this for 1.0: #2136 WDYT?

Reply:
makes sense yes